dhall-lsp-server/src/Dhall/LSP/Handlers/DocumentFormatting.hs

```diff
@@ -22,12 +23,11 @@ formatDocument fileUri _tabSize _insertSpaces = do
     let
       filePath = maybe (error "can't convert uri to file path") id $ J.uriToFilePath fileUri -- !FIXME: handle non-file uris
     txt <- lift $ Data.Text.IO.readFile filePath
-    case Formatting.formatDocument txt of 
+    case Formatting.formatDocument txt of
       (Right formatted) -> let
                              numLines = Data.Text.length txt
                              range = J.Range  (J.Position 0 0) (J.Position numLines 0)
                            in pure $ J.List [J.TextEdit range formatted]
       (Left err) -> do
                   lift $ LSP.Utility.logs $ "Error while formatting the document " <> show err
                   pure (J.List [])
-    
```

Comment:
I don't mind fixing trailing whitespace that much, but I disagree with removing newlines at the end of files. Maybe you should configure your editor to keep around the last newline?

Reply:
Restored the whitespace with f9deac6